src/lib/core/fiber.c

```diff
@@ -557,6 +557,15 @@ fiber_set_joinable(struct fiber *fiber, bool yesno)
 		fiber->flags &= ~FIBER_IS_JOINABLE;
 }
 
+void
+fiber_set_system(struct fiber *fiber, bool yesno)
```

Comment:
I don't think that anybody would ever want to clear this flag or set it after the fiber is stated. I think it's enough to pass it to `fiber_new_ex` in flags. For convenience, I'd add a helper to create a system fiber  - `fiber_new_system`.